### PR TITLE
Updated UIdocumentation.md

### PR DESCRIPTION
**Pull request by kayo11**
I updated the UIdocumentation.md to include new features and reviewed the previous version.

### CODE REVIEW

**Comment by kayo11** [comment]
@Ram81 right on it.

**Comment by kayo11** [comment]
@Ram81 Done, sorry for the grammar mistakes. I hope it's alright now.

File: docs/source/UIDocumentation.md [hunk 1]
```diff
@@ -1,160 +1,238 @@
...
+Frontend Design
+========
+## Architecture
+(All javascript files described here can be found in the /ide/static/js/ directory of Fabrik)
+
+#### ```addCommentModal.js```
+(The bullets contain methods specific to the file along with their explanations.)
+
+```addCommentModal.js``` is responsible for adding comments into the interface. It has three methods:
+* ```handleClick```
```
Comment (thatbrguy) [inline docs/source/UIDocumentation.md]: No explanation for this method? Having no explanation looks a bit weird while all other methods have them. Either move this method to the end of the list, or provide a brief explanation.

File: docs/source/UIDocumentation.md [hunk 2]
```diff
@@ -1,160 +1,238 @@
...
-* ```dismissAllErrors```
-* ```copyTrain``` - copy's train net for test
+* ```changeNetName``` - invoked on the event of net name changed; changes the name of the net
+* ```changeNetStatus``` - changes boolean for rebuilding.
+* ```changeNetPhase``` - changes the phase of the net
+* ```dismissError``` - dissmisses the error by ```errorIndex```
+* ```addError``` - adds error using ```errorText```
+* ```dismissAllErrors``` - dismisses all errors
+* ```copyTrain``` - copies the nets train option for the test option
 * ```trainOnly```
```
Comment (thatbrguy) [inline docs/source/UIDocumentation.md]: No explanation for this method? Having no explanation looks a bit weird while all other methods have them. Either move this method to the end of the list, or provide a brief explanation.

File: docs/source/UIDocumentation.md [hunk 3]
```diff
@@ -1,160 +1,238 @@
...
 
-After this, it calls ```getBetween``` to get the nodes between the x and y coordinates of the created line. 
+#### ```data.js```
+```data.js``` contains various different variables filled with data used throughout the application, it stores the layers
 
-The ```getBetween``` also serves the purpose of returning which direction in which the majority of the blocks between are. This is purely for performance, otherwise it would be seperated into a seperate function. 
+***
+#### ```error.js```
+```error.js``` contains Error React Component with two methods:
+* ```dismissError```
```
Comment (thatbrguy) [inline docs/source/UIDocumentation.md]: No explanation for this method? Having no explanation looks a bit weird while all other methods have them. Either move this method to the end of the list, or provide a brief explanation.

**Comment by thatbrguy** [comment]
@kayo11 Hello. Neat work! I noticed that some methods do not have explanations. Either move them to the end of the list, or provide a one liner explanation of what they do (recommended). I have marked some in my review. I have **may have missed** some methods so please check the entire file.

**Comment by kayo11** [comment]
@thatbrguy Yes, you're absolutely right. Thank you for the feedback. I'm on it

**Comment by kayo11** [comment]
@thatbrguy I think I got all of them. Should I change anything else?

File: docs/source/UIDocumentation.md [hunk 4]
```diff
@@ -1,160 +1,238 @@
...
+* ```updateLayerPosition``` - changes layer positon based on event
+* ```mouseUpEvent``` - invoked when mouse key is up
+* ```connectionEvent``` - invoked on connection; checks for cycles and modifies layers
+* ```detachConnectionEvent``` - modifies layers
+* ```render``` - renders canvas
+* ```drop``` - invoked on layer drop onto canvas; checks for errors
+
+```canvas.js``` also contains the code that decides whether a node's line needs to be rerouted if it is cutting through another node.
+
+#### ```canvas.js```'s placement algorithm
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: Can you change this to `visualisation algorithm` instead of `placement algorithm`

File: docs/source/UIDocumentation.md [hunk 5]
```diff
@@ -1,160 +1,238 @@
...
+* ```modifyLayer``` - modifies layer, passed in layer is the new layer, and layer id is the id it needs to be replaced at.
+* ```deleteLayer``` - deletes layers and removes inputs and outputs for it.
+* ```getLayerParameters``` - sums total amount of parameters and updates layer's parameters
+* ```calculateParameters``` - loops through net and invokes ```getLayerParameters```
+* ```adjustParameters``` - used to adjust layer parameters based on layer, change is passed in.
+* ```modifyLayerParams``` - modifies layer params based on layer and layerId, invoked by SetParams
 * ```loadLayerShapes``` - AJAXs to backend to model parameters
 * ```exportNet``` - AJAXs to backend and then passes back error/success
-  * framework
+* ```exportPrep``` - invoked by ```exportNet```; prepares for model export
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: use `preprocessed model object for export`

File: docs/source/UIDocumentation.md [hunk 6]
```diff
@@ -1,160 +1,238 @@
...
-  * connInfo
-  * originalEvent
-* ```drop```
-  * event
-```canvas.js``` also contains the code that decides whether a node's line needs to be rerouted if it is cutting through another node. 
-### ```canvas.js```'s placement algorithm
-The method it uses is the following:
-```checkIfCuttingLine``` is passed in a positional block that includes x and y coordinates (it assumes a px is at the end of each x and y) for each endpoint of the line it is checking. 
-Specifically, it is checking if the line formed with the coordinates in the positional block will cut into any other nodes between them.
+* ```changeNetName``` - invoked on the event of net name changed; changes the name of the net
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: not clear, can you rephrase this

File: docs/source/UIDocumentation.md [hunk 7]
```diff
@@ -1,160 +1,238 @@
...
+* ```changeNetStatus``` - changes boolean for rebuilding.
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: rephrase this

File: docs/source/UIDocumentation.md [hunk 8]
```diff
@@ -1,160 +1,238 @@
...
+* ```changeNetPhase``` - changes the phase of the net
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: be more specific to what phase

File: docs/source/UIDocumentation.md [hunk 9]
```diff
@@ -1,160 +1,238 @@
...
+* ```dismissError``` - dissmisses the error by ```errorIndex```
+* ```addError``` - adds error using ```errorText```
+* ```dismissAllErrors``` - dismisses all errors
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: You can leave some of the obvious methods

File: docs/source/UIDocumentation.md [hunk 10]
```diff
@@ -1,160 +1,238 @@
...
+* ```dismissError``` - dismisses opened error
+* ```render``` - renders the errors
+The error is passed in through props and then displayed to the user.
 
-After this, it calls ```getBetween``` to get the nodes between the x and y coordinates of the created line. 
+***
 
-The ```getBetween``` also serves the purpose of returning which direction in which the majority of the blocks between are. This is purely for performance, otherwise it would be seperated into a seperate function. 
+#### ```field.js```
+```field.js``` contains the various different fields used by the layer editor.
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: Fix the grammar

File: docs/source/UIDocumentation.md [hunk 11]
```diff
@@ -1,160 +1,238 @@
...
-```pane.js``` renders out all of the layers for selection by the user.
-### ```tabs.js```
-```tabs.js``` contains no methods and is used to switch between Train and Test layers. 
-### ```layer.js```
-```layer.js``` contains no methods. It is invoked by the ```canvas.js``` and it displays the actual layering on the jsplumb container. The position of each layer is set in the state of the layer. 
-### ```jsplumb.js```
-The ```jsplumb.js``` file contains code that handles the arrangement and the dragging/connecting of the layers. A new custom connector is created. There is an if function to check whether the node it is connecting is needing to be routed through an extension. A global variable stores this information, and the actual calculation is handled in ```checkIfCuttingNet``` and ```checkIfCuttingLine```. The global variable contains the amount of pixels it needs to move over, and it will contain direction it needs to go in (based on whether it is positive or 
-negative.)
+#### ```importTextbox.js```
+Renders the textbox for model import from text, using the three frameworks: Caffe, Keras and Tensorflow.
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: Rephrase this

File: docs/source/UIDocumentation.md [hunk 12]
```diff
@@ -1,160 +1,238 @@
...
+
+***
+
+#### ```jsplumb.js```
+The ```jsplumb.js``` file contains code that handles the arrangement and the dragging/connecting of layers. A new custom connector is created. There is an ```if``` function to check whether the node it is connecting to is needs to be routed through an extension. A global variable stores this information, and the actual calculation is handled in ```checkIfCuttingNet``` and ```checkIfCuttingLine```. The global variable contains the amount of pixels it needs to move over, and it will contain direction it needs to go in (based on whether it is positive or negative.)
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: Please rephrase this too, some of the lines are not clear

File: docs/source/UIDocumentation.md [hunk 13]
```diff
@@ -1,160 +1,238 @@
...
-* ```close```
-* ```trainOnly```
-    * e
-    
-### ```setParams.js``` 
-allows users to change layer parameters and reads paramters through ```data.js```
-### ```tooltip.js``` and ```tooltipData.js```
+
+#### ```netLayout_vertical.js```
+Defines the vertical align of network elements. ```netLayout_vertical.js``` uses a function, called ```allocatePosition``` to give the layers their position on the canvas. It finds the closest position available to the preferred position, and checks if a node is already on that position on the X axis (if one is, a position left or right is assigned). The file also uses a custom Topolical Sort to give the nodes there position (it is based on the inputs and outputs of nodes). Later it is checked if any nodes averlap and in case they do, they are moved.
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: replace `align` with `alignment`

**Comment by kayo11** [comment]
Now, I changed the code, according to your suggestions, and fixed some other issues. I hope everything is fine after my changes.

**Comment by kayo11** [comment]
@Ram81 @thatbrguy Is there anything else that needs a change?

File: docs/source/UIDocumentation.md [hunk 14]
```diff
@@ -1,160 +1,235 @@
...
 Both of these files contains code for tooltips and the ```tooltipData.js``` contains the actual tooltips that are rendered.
 
 note: tooltips are "hover" messages, they tell more information about something when a user hovers over an object.
- 
+***
+
+#### ```topbar.js```
+```topbar.js```  shows the top section of the sidebar. It has two methods:
+* ```checkURL``` - checks model import URL
+* ```render``` - renders the top section of the sidebar
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: @kayo11 as I said before no need to mention methods for which name of function clearly specifies the function

File: docs/source/UIDocumentation.md [hunk 15]
```diff
@@ -1,160 +1,235 @@
...
+#### ```setParams.js```
+Contains the following methods for the layer edition sidebar:
+* ```changeProps``` - changes layer properties
+* ```changeParams``` - changes layer parameters
+* ```close``` - closes sidebar
+* ```trainOnly``` - turns model into a train-only
+* ```handleKeyPress``` - handles layer delete after the 'delete' key is pressed
+* ```componentDidMount``` - invokes ```handleKeyPress```
+* ```componentWillUnmount``` - invokes ```handleKeyPress```
+* ```render``` - renders layer options sidebar
```
Comment (Ram81) [inline docs/source/UIDocumentation.md]: Please describe only methods defined by us, no need to describe standard react methods

**Comment by kayo11** [comment]
I deleted the more obvious functions but left out some that need explanation or are unclear and those which are particularly important.
